app/controllers/application_controller.rb

```diff
@@ -57,11 +58,42 @@ def no_matching_row e
       error: {
         code: "not_found",
         target: target,
-        message: e.message,
+        message: e.message
       }.compact
     }, status: :not_found
   end
 
+  def foreign_key_constraint_violation e
+    logger.debug "#{e}\n#{e.backtrace.join "\n"}"
```

Comment:
In my testing, the backtrace didn't propagate to the record_not_found method (it was nil)

Reply:
Oh, because you just created the exception with `new`, you didn't raise it.

Reply:
Right - but I can't reraise and have it hit the `not_found` exception handler ([see here](https://simonecarletti.com/blog/2009/11/re-raise-a-ruby-exception-in-a-rails-rescue_from-statement/) for an example). I'm not sure why a newly instantiated Exception object wouldn't have its backtrace set - that only gets set on `raise`??

Reply:
Anywho, I used your suggestion to have a `render_record_not_found` function that gets called from here and `record_not_found`, so that should resolve this issue.